specification/hooks.md

```diff
@@ -151,6 +151,12 @@ In languages with try/catch semantics, this means that exceptions thrown in `err
 
 > If an error occurs during the evaluation of `before` or `after` hooks, any remaining hooks in the `before` or `after` stages **MUST NOT** be invoked.
 
+#### Requirement 4.4.7
+
+> If an error occurs in the `before` hooks, the default value **MUST** be returned.
+
+Before hooks can impact evaluation by various means, such as mutating the `evaluation context`. Therefore, an error in the `before` hooks is considered abnormal execution, and the default should be returned.
+
```

Comment:
That makes sense for hooks that mutate the data. I suppose if it's written into the spec then anyone implementing a hook knows thath's the situation and it's easy to wrap in a try/catch. 👍

Reply:
The benefit of omitting `4.4.8` is that it allows consumers to be able to directly decide whether they want their provider to be called in Before hook error cases. A better way of achieving this (edge case?) behaviour might be to create an Error hook that calls the provider after a failed Before hook. I'm leaning towards including `4.4.8` and leaving it up to the Error hook.

Reply:
> The benefit of omitting 4.4.8 is that it allows consumers to be able to directly decide whether they want their provider to be called in Before hook error cases.

I think it really allows the SDK developer to decide that, unless I'm misunderstanding.

> A better way of achieving this (edge case?) behaviour might be to create an Error hook that calls the provider after a failed Before hook. I'm leaning towards including 4.4.8 and leaving it up to the Error hook.

That's an interesting idea... if there is some side-effect/impression data the application author REALLY wants even in the case of an error in the `before` hook, this could be a solution.

Ya... I'm starting to believe maybe adding the 4.4.8 point could be a good idea.